### PR TITLE
Add Xcode 16 Beta and Swift 6 support

### PR DESCRIPTION
**Pull request by Tyler-Keith-Thompson**
This PR attempts to have Afluent run in Swift6 mode on the Xcode 16 beta, but still run on Xcode 15 with 5.10. 

Simultaneously, it must handle strict concurrency, so it's a bit of a noisy change.

### CODE REVIEW

File: Tests/AfluentTests/WorkerTests/ShareTests.swift [hunk 1]
```diff
@@ -12,8 +12,8 @@ import Testing
 
 struct ShareTests {
     @Test func unsharedTaskExecutesRepeatedly() async throws {
-        try await withMainSerialExecutor {
-            try await confirmation(expectedCount: 4) { exp in
+        try await confirmation(expectedCount: 4) { exp in
```
Comment (Tyler-Keith-Thompson) [inline Tests/AfluentTests/WorkerTests/ShareTests.swift:15]: https://github.com/pointfreeco/swift-concurrency-extras/issues/27

**Comment by aim2120** [comment]
Hey, sorry for the delay on reviewing this. My goal is to give it a thorough review before Friday.

File: Tests/AfluentTests/WorkerTests/ShareTests.swift (hunk 1 above)
Comment (aim2120) [reply Tests/AfluentTests/WorkerTests/ShareTests.swift:15]: [It looks like they merged the fix for this](https://github.com/pointfreeco/swift-concurrency-extras/pull/28)! Not sure if you want to pull in a SHA to get it, but just wanted to update the thread.

**Comment by Tyler-Keith-Thompson** [comment]
@aim2120 I'm happy with where this is at now. I'm going to go ahead and merge. If you do have time to check it out, please feel free to retroactively leave any comments.

**Pull request merged**